iroh/src/node/rpc.rs

```diff
@@ -209,22 +210,47 @@ impl<D: BaoStore> Handler<D> {
         use gossip::Request::*;
         match msg {
             Subscribe(msg) => {
-                chan.bidi_streaming(msg, self, |handler, req, updates| {
-                    handler.inner.gossip_dispatcher.subscribe_with_opts(
-                        req.topic,
-                        iroh_gossip::dispatcher::SubscribeOptions {
-                            bootstrap: req.bootstrap,
-                            subscription_capacity: req.subscription_capacity,
-                        },
-                        Box::new(updates),
-                    )
+                chan.try_server_streaming(msg, self, |handler, req| async move {
+                    let sub = handler
+                        .inner
+                        .gossip
+                        .join_and_subscribe(req.topic, req.bootstrap)
+                        .await?;
+                    let stream = BroadcastStream::new(sub)
+                        .map(|res| res.map_err(|e| RpcError::from(anyhow::anyhow!("{:?}", e))));
+                    Ok(stream)
                 })
                 .await
             }
-            Update(_msg) => Err(RpcServerError::UnexpectedUpdateMessage),
+            Broadcast(msg) => chan.rpc(msg, self, Self::gossip_broadcast).await,
+            BroadcastNeighbours(msg) => {
+                chan.rpc(msg, self, Self::gossip_neighbours_broadcast).await
+            }
+            Quit(msg) => chan.rpc(msg, self, Self::gossip_quit).await,
         }
     }
 
+    async fn gossip_broadcast(self, msg: gossip::BroadcastRequest) -> RpcResult<()> {
+        self.inner.gossip.broadcast(msg.topic, msg.message).await?;
```

Comment:
Does this even work if I am not subscribed to the topic?

The usage of the client API allows calling broadcast without subscribing to a topic, as far as I can see.